src/wallet/builder.rs

```diff
@@ -316,11 +321,34 @@ impl OrdTransactionBuilder {
             txid: args.input.id,
             vout: args.input.index,
         };
+
         // tx out
+        #[cfg(feature = "rune")]
```

Comment:
This feature-based logic inside a method seems like something impossible to support in future, and would probably not work correctly for some use cases.

What's even more confusing is that the type is called `RevealTransactionBuilder` but we build a runestone transaction instead.

I think if we want to use this crate in future, it would be better to invest some time to refactor the code to be less confusing and more correct.

Reply:
@Maximkaaa Actually etching requires an inscription. I agree with your point, so at least I've created a dedicated function for this in the rune module.